dags/microsoft_store.py

```diff
@@ -0,0 +1,120 @@
+from airflow.providers.cncf.kubernetes.secret import Secret
+from airflow import DAG
+from airflow.sensors.external_task import ExternalTaskMarker
+from airflow.sensors.external_task import ExternalTaskSensor
+from airflow.utils.task_group import TaskGroup
+import datetime
+from operators.gcp_container_operator import GKEPodOperator
+from utils.constants import ALLOWED_STATES, FAILED_STATES
+from utils.gcp import bigquery_etl_query, bigquery_dq_check
+ 
+# Deploy value associated with Microsoft Store keys in k8s secret `airflow-gke-secrets` in environments Microsoft variables
+microsoft_client_id = Secret(deploy_type="env", deploy_target="MICROSOFT_CLIENT_ID", secret="airflow-gke-secrets", key="MICROSOFT_CLIENT_ID")
+microsoft_client_secret = Secret(deploy_type="env", deploy_target="MICROSOFT_CLIENT_SECRET", secret="airflow-gke-secrets", key="MICROSOFT_CLIENT_SECRET")
+microsoft_tenant_id = Secret(deploy_type="env", deploy_target="MICROSOFT_TENANT_ID", secret="airflow-gke-secrets", key="MICROSOFT_TENANT_ID")
+microsoft_store_app_list =  Secret(deploy_type="env", deploy_target="MICROSOFT_STORE_APP_LIST", secret="airflow-gke-secrets", key="MICROSOFT_STORE_APP_LIST")
+
+docs = """
+
+#### Owner
+mhirose@mozilla.com
+#### Tags
+* impact/tier_2
+* repo/bigquery-etl
+"""
+
+default_args = {
+    "owner": "mhirose@mozilla.com",
+    "start_date": datetime.datetime(2024, 6, 18, 0, 0),
+    "end_date": None,
+    "email": ["telemetry-alerts@mozilla.com", "mhirose@mozilla.com"],
+    "depends_on_past": False,
+    "retry_delay": datetime.timedelta(seconds=1800),
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+}
+
+read_gke_secret = GKEPodOperator(
```

Comment:
No need to have this. What you actually need is to have `secrets=[<secret_names>]` in each of your DAG's tasks

Reply:
Check out this example:
https://github.com/mozilla/telemetry-airflow/blob/1ebe9a9de2245d468c63b701556830ec0f0c53ce/dags/glam_glean_imports.py#L127

Reply:
I think I've added it correctly. Thank you!